### DISCUSSION TITLE
error "non-concrete value struct" when trying to look on cue struct

### DISCUSSION THREAD

**Discussion by ying-jeanne** [cuelang/cue#985] [Q&A]
the cue version 0.3.2

When the inputdef.IncompleteKind() is cue.StructKind, i would like to loop on all fields of inputdef by using function iter, err := inputdef.Fields(cue.Optional(true), cue.ResolveReferences(true), cue.Concrete(false)).  

Then I got an error  panels.268435455: non-concrete value struct, knowing that i would like to loop on element not necessarily concrete. What potentially is wrong here?

**Comment by myitcv** [comment]
Please can you share a repro of what you are seeing here?

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/985.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
